core/string/ustring.cpp

```diff
@@ -4306,6 +4306,96 @@ String String::replace_first(const char *p_key, const char *p_with) const {
 	return *this;
 }
 
+String String::replace_char(char32_t p_key, char32_t p_with) const {
+	ERR_FAIL_COND_V_MSG(p_with == 0, String(), "`with` must not be null.");
+
+	int len = length();
+	if (p_key == 0 || len == 0) {
```

Comment:
Technically `0` is a valid UTF32 character. Do we want to check for `p_key == 0` explicitly? It may bite us later...

Reply:
The `0` character won't be in any valid string currently, only as the trailing null, and replacing that would be bad I'd say

And I'd say that `0` here is equivalent to `p_key == ""` for the normal replace method, where we exit

Reply:
Ok, that's fine. I guess making the code "future proof" for a goal there's no consensus over doesn't really make sense. Feel free to resolve this one.

Reply:
There's work or discussion on adding nulls internally but currently I'll leave it